packages/tpetra/core/src/Tpetra_Details_copyConvert.hpp

```diff
@@ -279,10 +279,13 @@ namespace { // (anonymous)
         // it's cheaper to allocate.  Hopefully users aren't doing
         // aliased copies in a tight loop.
         auto src_copy = Kokkos::create_mirror (Kokkos::HostSpace (), src);
+        // DEEP_COPY REVIEW - NOT TESTED
```

Comment:
In my opinion, this logic is probably wrong. The intention of this copy code is a safe utility function to avoid possible pointer aliasing. However, this logic does not solve the issue anyway. First, the aliasing only consider that the view is contiguous. If two views are subviewed and strided. they can be copied without a trouble. Second, this logic creates a mirror view to host for temporal memory. This probably work on a gpu but it is more expensive. It can just use device memory instead. Also, if the default space is host, then this logic is wrong as the mirror view construction is just a softcopy and src_copy is still overlapping with dst. I do not recommend to use this function if it is used somewhere.